platform/admin/core/admin-core-configpolicy/src/main/java/org/codice/ddf/admin/config/policy/AdminConfigPolicy.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.config.policy;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.shiro.authz.Permission;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+import ddf.security.permission.CollectionPermission;
+import ddf.security.permission.KeyValueCollectionPermission;
+import ddf.security.permission.KeyValuePermission;
+import ddf.security.policy.extension.PolicyExtension;
+
+public class AdminConfigPolicy implements PolicyExtension {
+
+    public static final String FEATURE_NAME = "feature.name";
+
+    public static final String SERVICE_PID = "service.pid";
+
+    public static final String VIEW_FEATURE_ACTION = "view-" + FEATURE_NAME;
+
+    public static final String VIEW_SERVICE_ACTION = "view-" + SERVICE_PID;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdminConfigPolicy.class);
+
+    Map<String, List<KeyValueCollectionPermission>> featurePolicyPermissions = new HashMap<>();
+
+    Map<String, List<KeyValueCollectionPermission>> servicePolicyPermissions = new HashMap<>();
+
+    @Override
+    public KeyValueCollectionPermission isPermittedMatchAll(
+            CollectionPermission subjectAllCollection,
+            KeyValueCollectionPermission matchAllCollection) {
+        return isPermittedMatchOne(subjectAllCollection, matchAllCollection);
+    }
+
+    @Override
+    public KeyValueCollectionPermission isPermittedMatchOne(
+            CollectionPermission subjectAllCollection,
+            KeyValueCollectionPermission matchOneCollection) {
+
+        if (matchOneCollection.getAction() == null || (!matchOneCollection.getAction()
+                .equals(VIEW_FEATURE_ACTION) && !matchOneCollection.getAction()
+                .equals(VIEW_SERVICE_ACTION))) {
+            return matchOneCollection;
+        }
+
+        KeyValueCollectionPermission newMatchCollection = new KeyValueCollectionPermission();
+
+        for (Permission permission : matchOneCollection.getPermissionList()) {
+
+            if (!(permission instanceof KeyValuePermission)) {
+                continue;
```

Comment:
Maybe I am reading this incorrectly but I think all of these continues are wrong and will basically make this policy grant any permission that is continued on.  I would suggest instead initializing `newMatchCollection` with all permissions and then filter out only permitted service permissions.  All other permissions or denied service permissions should remain.

https://github.com/codice/ddf/blob/ddf-2.9.0/platform/security/platform-security-core-api/src/main/java/ddf/security/policy/extension/PolicyExtension.java#L47-L48